src/rules/no-namespace.js

```diff
@@ -16,13 +16,150 @@ module.exports = {
+          variable.defs[0].node === node
+        )
+        const namespaceReferences = namespaceVariable.references
+        const namespaceIdentifiers = namespaceReferences.map(reference => reference.identifier)
+        const canFix = namespaceIdentifiers.length && !usesNamespaceAsObject(namespaceIdentifiers)
+
+        context.report({
+          node,
+          message: `Unexpected namespace import.`,
+          fix: canFix && (fixer => {
+            const scopeManager = context.getSourceCode().scopeManager
+            const fixes = []
+
+            // Pass 1: Collect variable names that are already in scope for each reference we want
+            // to transform, so that we can be sure that we choose non-conflicting import names
+            const importNameConflicts = {}
+            namespaceIdentifiers.forEach((identifier) => {
+              const parent = identifier.parent
+              if (parent && parent.type === 'MemberExpression') {
+                const importName = getMemberPropertyName(parent)
+                const localConflicts = getVariableNamesInScope(scopeManager, parent)
+                if (!importNameConflicts[importName]) {
+                  importNameConflicts[importName] = localConflicts
+                } else {
+                  localConflicts.forEach((c) => importNameConflicts[importName].add(c))
+                }
+              }
+            })
+
+            // Choose new names for each import
+            const importNames = Object.keys(importNameConflicts)
+            const importLocalNames = generateLocalNames(
+              importNames,
+              importNameConflicts,
+              namespaceVariable.name
+            )
+
+            // Replace the ImportNamespaceSpecifier with a list of ImportSpecifiers
+            const namedImportSpecifiers = importNames.map((importName) =>
+              importName === importLocalNames[importName]
+                ? importName
+                : `${importName} as ${importLocalNames[importName]}`
+            )
+            fixes.push(fixer.replaceText(node, `{ ${namedImportSpecifiers.join(', ')} }`))
+
+            // Pass 2: Replace references to the namespace with references to the named imports
+            namespaceIdentifiers.forEach((identifier) => {
+              const parent = identifier.parent
+              if (parent && parent.type === 'MemberExpression') {
+                const importName = getMemberPropertyName(parent)
+                fixes.push(fixer.replaceText(parent, importLocalNames[importName]))
+              }
+            })
+
+            return fixes
+          }),
+        })
       },
     }
   },
 }
+
+/**
+ * @param {Identifier[]} namespaceIdentifiers
+ * @returns {boolean} `true` if the namespace variable is more than just a glorified constant
+ */
+function usesNamespaceAsObject(namespaceIdentifiers) {
+  for (let i = 0; i < namespaceIdentifiers.length; i++) {
```

Comment:
can this be written as a `.some` instead of a for loop?

Reply:
Sure thing. ✅